pytorch_lightning/core/optimizer.py

```diff
@@ -186,16 +187,29 @@ def dis_closure():
             make_optimizer_step = not self._should_accumulate
 
         trainer = self._trainer
+        model = trainer.get_model()
         optimizer = self._optimizer
 
         if make_optimizer_step:
+
+            if self._optimizer_idx is None:
+                for opt_idx, opt in enumerate(trainer.optimizers):
+                    if self == opt:
+                        self._optimizer_idx = opt_idx
+                        break
+
+            ddp_plugin = trainer.accelerator_backend.ddp_plugin
+            if ddp_plugin.using_rpc_async:
```

Comment:
Again, should only do this if ddp_plugin is an RPCPlugin

Reply:
Same here.